### PR TITLE
Chore/arm 191 hook tests

### PR DESCRIPTION
**Pull request by judithrocketmakers**
## What's new?

- Adds Jest hook tests to all hooks where this was viable
  - Note: Other hooks will either need to be tested with wrapper stories or in components.
- Renames useIsHovering to useIsHovered to match focus equivalent.

## Ticket number(s) in JIRA (if internal)

ARM-191

[board](https://rocketmakers.atlassian.net/jira/software/projects/ARM/boards/154)

## Checklist

- [ ] are your changes in Storybook?
- [ ] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [ ] does _everything_ have jsdoc?
- [ ] is everything exported from index.ts?
- [ ] are all new hooks added to `src/hooks/hooks.stories.mdx` or given their own docs in Storybook?
- [ ] are all new SCSS mixins added to `src/theme/mixins.stories.mdx`?
- [ ] are all new SCSS variables added to `src/theme/variables.stories.mdx`?

### CODE REVIEW

File: module/src/hooks/form/form.utils.ts [hunk 1]
```diff
@@ -75,7 +78,7 @@ export function valueByKeyChain<TData, TValue>(state: TData, keyChain: KeyChain)
  * @returns {boolean} `true` if the item passed conforms to the binding props interface `IBindingProps` else `false`. Also casts if `true`.
  */
 export function isBindingProps<TValue>(item?: any): item is IBindingProps<TValue> {
-  return !!item?.setValue && !!item?.dispatch && !!item?.keyChain && !!item?.myValidationErrors;
+  return !!item && 'setValue' in item && 'dispatch' in item && 'keyChain' in item && 'myValidationErrors' in item;
```
Comment (jlews) [inline module/src/hooks/form/form.utils.ts:81]: Probably revert for consistency. Also, using `in` like this removes all type safety on the properties being accessed, so if `setValue` ever got renamed for example, the compiler would just let this through and a bug would be introduced with no warning.

Comment (judithrocketmakers) [reply module/src/hooks/form/form.utils.ts]: Fixed as discussed.

File: module/src/hooks/form/form.stories.mdx [hunk 2]
```diff
@@ -97,7 +97,7 @@ Initial state can also be passed as a React callback. The callback will be execu
 
 This is extremely useful when a form relies on remote data that could update at any time, it allows the form to update with the latest changes from the server without losing the user's current edits by completely replacing the form data with the remote update.
 
-**NOTE:** When using an initial data function, be aware that it will be executed to provide initial data whenever it's reference updates. It's essential therefore that any initial data function is a callback with a carefully selected set of dependencies, rather than an anonymous function or constant function.
+**NOTE:** When using an initial data function, be aware that it will be executed to provide initial data whenever its reference updates. It's essential therefore that any initial data function is a callback with a carefully selected set of dependencies, rather than an anonymous function or constant function.
```
Comment (SheaMurphy) [inline module/src/hooks/form/form.stories.mdx:100]: Hero

Comment (judithrocketmakers) [reply module/src/hooks/form/form.stories.mdx:100]: I can't help it, I'm a pedant over apostrophes...

**Review by tomhinsley — APPROVED** [review]
Tests running locally, nothing jumping out as untoward

**Pull request merged**